src/main/java/com/cgdp/library/service/BookService.java

```diff
@@ -0,0 +1,35 @@
+package com.cgdp.library.service;
+
+
+import com.cgdp.library.dto.author.AuthorDTO;
+import com.cgdp.library.dto.book.BookDTO;
+import com.cgdp.library.dto.book.CreateBookRequestDTO;
+import com.cgdp.library.entity.AuthorEntity;
+import com.cgdp.library.entity.BookEntity;
+import com.cgdp.library.repository.AuthorRepository;
+import com.cgdp.library.repository.BookRepository;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class BookService {
+
+    private final BookRepository bookRepository;
+    private final AuthorRepository authorRepository;
+    private final ModelMapper modelMapper;
+
+    public BookDTO createBook(CreateBookRequestDTO requestDTO) {
+
+        AuthorEntity authorEntity = authorRepository.findById(requestDTO.getAuthorId())
```

Comment:
It's very inefficient to do query to get the author in order to create a book, we should have a method just to check if it exists, if not throw exception (create custom exception, something like `NotFoundException`)

Also check this https://www.baeldung.com/jpa-entity-manager-get-reference